packages/@webex/plugin-meetings/src/reachability/index.ts

```diff
@@ -395,16 +409,196 @@ export default class Reachability {
     });
   }
 
+  /**
+   * Returns true if we've obtained all the reachability results for all the public clusters
+   * In other words, it means that all public clusters are reachable over each protocol,
+   * because we only get a "result" if we managed to reach a cluster
+   *
+   * @returns {boolean}
+   */
+  private areAllPublicClusterResultsReady() {
+    return isEqual(this.expectedResultsCount.public, this.resultsCount.public);
+  }
+
+  /**
+   * Returns true if we've obtained all the reachability results for all the clusters
+   *
+   * @returns {boolean}
+   */
+  private areAllResultsReady() {
+    return isEqual(this.expectedResultsCount, this.resultsCount);
+  }
+
+  /**
+   * Resolves the promise returned by gatherReachability() method
+   * @returns {void}
+   */
+  private resolveReachabilityPromise() {
+    if (this.vmnTimer) {
+      clearTimeout(this.vmnTimer);
+    }
+    if (this.publicCloudTimer) {
+      clearTimeout(this.publicCloudTimer);
+    }
+
+    this.logUnreachableClusters();
+    this.reachabilityDefer?.resolve();
+  }
+
+  /**
+   * Aborts all cluster reachability checks that are in progress
+   *
+   * @returns {void}
+   */
+  private abortClusterReachability() {
+    Object.values(this.clusterReachability).forEach((clusterReachability) => {
+      clusterReachability.abort();
+    });
+  }
+
+  /**
+   * Sends a metric with all the statistics about how long reachability took
+   *
+   * @returns {void}
+   */
+  private async sendMetric() {
+    const getStatistics = (results, protocol, isVideoMesh) => {
```

Comment:
nitpick: might be better to extract as separate func, that way testing is easier

Reply:
Ok, I've done that, but both `getStatistics()` and `sendMetric()` are private so they shouldn't be called from the tests anyway. I've worked around this by declaring them as `protected` so in the test I created a class that extends `Reachability` class and added public methods that call `getStatistics()` and `sendMetric()` to allow us to test them. I've added more unit tests for these methods now, I agree it makes it easier to test all the edge cases, but I feel that it makes the unit tests too closely coupled to the implementation, which then makes the tests not very useful if you try to refactor something inside `Reachability` class, so it's a trade off....

Reply:
I missed that it was a private function. 
Yeah, I see your point and I agree, but the effort for still doing it is very appreciated!